WalletWasabi.Fluent/ViewModels/ApplicationViewModel.cs

```diff
@@ -67,11 +67,37 @@ private IObservable<bool> AboutCanExecute()
 
 	public void OnShutdownPrevented(bool restartRequest)
 	{
+		if (!MainViewCanShutdown())
+		{
+			MainViewModel.Instance.ShowDialogAlert();
+			return;
+		}
```

Comment:
This should be placed after 
```c#
MainViewModel.Instance.ApplyUiConfigWindowState(); // Will pop the window if it was minimized.
```
Because the window should be brought to the front when it is minimized.

BTW question... Should we block the user when they try to close the software from the tray?

Reply:
Fixed.

Not sure about blocking tray exit, guess we should be consistent and block all exits ?